crates/solver-service/src/apis/quote/generation.rs

```diff
@@ -653,16 +651,23 @@ impl QuoteGenerator {
 		Ok(order)
 	}
 
-	/// Compute the orderIdentifier for an EIP-3009 order by building a StandardOrder
-	/// and calling the contract's orderIdentifier function - same as the intents script
+	/// Compute the orderIdentifier for an EIP-3009 order using pure Rust implementation.
 	/// Returns (nonce, order_identifier)
+	///
+	/// This uses the same algorithm as StandardOrderType.sol's orderIdentifier function
+	/// but computes it directly in Rust without any external contract calls or `cast` commands.
 	fn compute_eip3009_order_identifier(
 		&self,
 		request: &GetQuoteRequest,
```

Comment:
I'm pretty sure we have (or should have) a request.order -> StandardOrder From conversion somewhere we can re-use.